internal/blockdb/collect_test.go

```diff
@@ -0,0 +1,124 @@
+package blockdb
+
+import (
+	"context"
+	"errors"
+	"strconv"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+type mockTxFinder func(ctx context.Context, height uint64) ([][]byte, error)
+
+func (f mockTxFinder) FindTxs(ctx context.Context, height uint64) ([][]byte, error) {
+	return f(ctx, height)
+}
+
+type mockBlockSaver func(ctx context.Context, height int, txs [][]byte) error
+
+func (f mockBlockSaver) SaveBlock(ctx context.Context, height int, txs [][]byte) error {
+	return f(ctx, height, txs)
+}
+
+func TestCollector_Collect(t *testing.T) {
```

Comment:
Slightly preferable to have the outer test and/or its subtests can use `t.Parallel` since these are exercising asychronous communication across goroutines, even with the nanosecond duration.

Reply:
Yup, I need to get in the habit of that. Go tests run so fast (with caching) that I often forget about t.Parallel.